dspace-spring-rest/src/main/java/org/dspace/app/rest/converter/query/SearchQueryConverter.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.converter.query;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.dspace.app.rest.model.query.RestSearchOperator;
+import org.dspace.app.rest.parameter.SearchFilter;
+
+public class SearchQueryConverter {
```

Comment:
We need to get in the habit of creating Javadocs for all public classes & methods. It makes it much easier to understand the code.  This class needs JavaDocs and so does its methods.

Reply:
It doesn't only help others; I've lost count of the number of times I was able to solve a design problem only by writing documentation in advance, or caught an error while trying to document code I had just written.